shared/actions/profile.js

```diff
@@ -412,18 +428,178 @@ function backToProfile (): AsyncAction {
   }
 }
 
+type PgpInfoError = {
+  errorText: ?string,
+  errorEmail1: boolean,
+  errorEmail2: boolean,
+  errorEmail3: boolean,
+}
+
+function generatePgp (): GeneratePgp {
+  return {
+    type: Constants.generatePgp,
+    payload: undefined,
+  }
+}
+
+function dropPgp (kid: KID): DropPgp {
+  return {
+    type: Constants.dropPgp,
+    payload: {kid},
+  }
+}
+
+// This can be replaced with something that makes a call to service to validate
+function checkPgpInfoForErrors (pgpInfo: PgpInfo): PgpInfoError {
+  const errorEmail1 = (pgpInfo.email1 && isValidEmail(pgpInfo.email1))
+  const errorEmail2 = (pgpInfo.email2 && isValidEmail(pgpInfo.email2))
+  const errorEmail3 = (pgpInfo.email3 && isValidEmail(pgpInfo.email3))
+
+  return {
+    errorText: isValidName(pgpInfo.fullName) || errorEmail1 || errorEmail2 || errorEmail3,
```

Comment:
but you want the error text still, like the errorEmail1 could be `there isn't a @" so you want the error text to show an error for the valid name or the error for the email 1 ...